frontend/sw360-portlet/src/main/java/org/eclipse/sw360/portal/portlets/projects/ProjectPortlet.java

```diff
@@ -341,11 +339,21 @@ private void createClearingRequest(ResourceRequest request, ResourceResponse res
             clearingRequest = OBJECT_MAPPER.convertValue(crNode, ClearingRequest.class);
             clearingRequest.setRequestingUser(user.getEmail());
             clearingRequest.setClearingState(ClearingRequestState.NEW);
+            ModerationService.Iface modClient = thriftClients.makeModerationClient();
+            Integer criticalCount = modClient.getCriticalClearingRequestCount();
+            clearingRequest.setPriority(criticalCount > 1 ? null : clearingRequest.getPriority());
             LiferayPortletURL projectUrl = createDetailLinkTemplate(request);
             projectUrl.setParameter(PROJECT_ID, clearingRequest.getProjectId());
             projectUrl.setParameter(PAGENAME, PAGENAME_DETAIL);
             ProjectService.Iface client = thriftClients.makeProjectClient();
-            requestSummary = client.createClearingRequest(clearingRequest, user, projectUrl.toString());
+            Integer dateLimit = ModerationPortletUtils.loadPreferredClearingDateLimit(request, user);
+            dateLimit = (ClearingRequestPriority.CRITICAL.equals(clearingRequest.getPriority()) && criticalCount < 2) ? 0 : (dateLimit < 1) ? 7 : dateLimit;
+            if (!SW360Utils.isValidDate(clearingRequest.getRequestedClearingDate(), DateTimeFormatter.ISO_LOCAL_DATE, dateLimit)) {
```

Comment:
null check for date? clearingRequest.getRequestedClearingDate().

Reply:
`requestedClearingDate` is required field in thrift layer, It can't be null.
https://github.com/eclipse/sw360/blob/a91c7a9f6aab87e3915b320bfad8f763ad06d734/libraries/lib-datahandler/src/main/thrift/projects.thrift#L212